src/Concerns/SendsRequests.php

```diff
@@ -83,6 +88,112 @@ protected function formattedReverseQuery($query)
         return $query;
     }
 
+    /**
+     * Format coordinate as string for GET requests.
+     */
+    protected function formatCoordinateAsString(Coordinate|string|array $coordinate): string
+    {
+        if ($coordinate instanceof Coordinate) {
+            return $coordinate->toString();
+        }
+
+        if (is_string($coordinate)) {
+            // Validate it's a coordinate format, not an address
+            if ($this->isCoordinateString($coordinate)) {
+                return $coordinate;
+            }
+
+            // Return as-is for addresses
+            return $coordinate;
+        }
+
+        // Array format: [lat, lng] or [lat, lng, id]
+        return Coordinate::fromArray($coordinate)->toString();
+    }
+
+    /**
+     * Format coordinate as object for POST requests.
+     *
+     * Returns either:
+     * - array{lat: float, lng: float, id?: string} for coordinates
+     * - string for addresses (passed through)
+     *
+     * @return array{lat: float, lng: float, id?: string}|string
+     */
+    protected function formatCoordinateAsObject(Coordinate|string|array $coordinate): array|string
+    {
+        if ($coordinate instanceof Coordinate) {
+            return $coordinate->toArray();
+        }
+
+        if (is_string($coordinate)) {
+            // Check if it's a coordinate string
+            if ($this->isCoordinateString($coordinate)) {
+                return Coordinate::fromString($coordinate)->toArray();
+            }
+
+            // Return as-is for addresses
+            return $coordinate;
+        }
+
+        // Array format: [lat, lng] or [lat, lng, id]
+        return Coordinate::fromArray($coordinate)->toArray();
+    }
+
+    /**
+     * Check if a string looks like a coordinate (lat,lng or lat,lng,id).
+     *
+     * Validates that:
+     * - At least 2 comma-separated parts exist
+     * - First two parts are numeric (the coordinate values)
+     * - Values are within valid geographic ranges
+     */
+    protected function isCoordinateString(string $value): bool
+    {
+        $parts = explode(',', $value);
+
+        if (count($parts) < 2) {
+            return false;
+        }
+
+        $lat = trim($parts[0]);
+        $lng = trim($parts[1]);
+
+        if (! is_numeric($lat) || ! is_numeric($lng)) {
+            return false;
+        }
+
+        $latFloat = (float) $lat;
+        $lngFloat = (float) $lng;
+
+        // Validate lat/lng ranges
+        return $latFloat >= -90 && $latFloat <= 90
+            && $lngFloat >= -180 && $lngFloat <= 180;
+    }
+
+    /**
+     * Normalize distance mode value, mapping haversine to straightline for backward compatibility
+     */
+    protected function normalizeDistanceMode(string|DistanceMode $mode): string
+    {
+        $value = $mode instanceof DistanceMode ? $mode->value : $mode;
+
+        // Map haversine → straightline for backward compatibility
+        return $value === 'haversine' ? 'straightline' : $value;
```

Comment:
I don't think this conversion is necessary. The API supports both haversine and straightline AFAIK